### PR TITLE
add xamarin files from v1 repo

### PR DESCRIPTION
**Pull request by qooroo**

### CODE REVIEW

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/ReactiveTrader.cs [hunk 1]
```diff
@@ -1,49 +1,37 @@
...
+        public void Initialize(string username, string[] servers, ILoggerFactory loggerFactory = null, string authToken = null)
         {
             _loggerFactory = loggerFactory ?? new DebugLoggerFactory();
             _log = _loggerFactory.Create(typeof(ReactiveTrader));
-            _connectionProvider = new ConnectionProvider(username, servers, _loggerFactory);
             var concurrencyService = new ConcurrencyService();
 
             _serviceClientContainer = new WampServiceClientContainer(servers[0], username, concurrencyService, _loggerFactory);
-            _serviceClientContainer.ConnectAsync().Wait();
+            _serviceClientContainer.ConnectAsync();
```
Comment (qooroo) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/ReactiveTrader.cs]: Yeah ideally we'd await this, but might be a pain from the initialize call chain. Is this fire and forget good enough?

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ObservableCallback.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 using Adaptive.ReactiveTrader.Shared.Logging;
 using System;
+using System.Diagnostics;
```
Comment (qooroo) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ObservableCallback.cs]: unused?

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs [hunk 3]
```diff
@@ -93,7 +93,7 @@ public static class ServiceObservableExtensions
                 disposables.Add(
                     source.Subscribe(innerSource =>
                     {
-                        IObservable<TValue> innerSourcePublished = innerSource.Replay(1).RefCount();
+                        IObservable<TValue> innerSourcePublished = innerSource.Publish().RefCount();
```
Comment (qooroo) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs]: this method is called `ToLastValueObservableDictionary` - if we change the behaviour, should we also change the method name?

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/WampConnection.cs [hunk 4]
```diff
@@ -39,46 +39,47 @@ public WampConnection(string serverUri, string userName, ILoggerFactory loggerFa
...
                 _openCalled = true;
                 _log.Info("Opening Connection");
 
                 _disposables.Add(SubscribeToConnectionStatus());
-
-                await _channel.Open().ConfigureAwait(false);
+                return _channel.Open();
             }
+
+            return Task.Delay(0);
```
Comment (qooroo) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/WampConnection.cs]: prefer `Task.CompletedTask`

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/ReactiveTrader.cs (hunk 1 above)
Comment (UncleFirefox) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/ReactiveTrader.cs]: I think for now we'll reintroduce the wait and focus on awaitability in another PR!

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ObservableCallback.cs (hunk 2 above)
Comment (UncleFirefox) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ObservableCallback.cs]: Done!

File: src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs (hunk 3 above)
Comment (UncleFirefox) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs]: Honestly, I don't know, it was coming from the last Xamarin changes in James' branch. Would it make a difference?

Comment (qooroo) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs]: @JPWatson any memories of this?

Comment (JPWatson) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs]: Think you should roll back, it changes the semantics. It was probably a change I made in the middle of the night to try and get the thing working.

Comment (UncleFirefox) [inline src/xamarin-client/Adaptive.ReactiveTrader.Client.Domain/Transport/Wamp/ServiceObservableExtensions.cs]: Done!

**Pull request merged**